tests/proxying.js

```diff
@@ -442,3 +443,49 @@ test('Proxying() - proxy to a non existing server - GET request will error - sho
 
     t.end();
 });
+
+test('Proxying() - Trailer header - 400s when Trailer header is present', async (t) => {
+    const server = new HttpServer();
+    server.request = reqFn;
+    const serverAddr = await server.listen();
+
+    const proxy = new ProxyServer(
+        [
+            {
+                name: 'foo',
+                proxy: {
+                    a: `${serverAddr}/some/path`,
+                },
+                version: '1.0.0',
+                content: '/',
+            },
+        ],
+        { name: 'mylayout' },
+    );
+
+    const proxyAddr = await proxy.listen();
+
+    const { stdout } = await new Promise((resolve, reject) => {
+        exec(
```

Comment:
I was unable to use the `request` test helper as that would throw an error if the trailer header was present and requests never made it to the server. I did test out the experimental undici based built in fetch module in node 17.5 and that did not have this issue so undici would likely be another way to make this request. Exec and curl work fine too.